service/security/shiro/src/main/java/org/eclipse/kapua/service/authorization/role/shiro/RoleServiceImpl.java

```diff
@@ -164,8 +164,10 @@ public Role update(Role role) throws KapuaException {
             if (roleRepository.countOtherEntitiesWithNameInScope(tx, role.getScopeId(), role.getId(), role.getName()) > 0) {
                 throw new KapuaDuplicateNameException(role.getName());
             }
+            current.setName(role.getName());
+            current.setName(role.getDescription());
             // Do update
-            return roleRepository.update(tx, current, role);
+            return roleRepository.update(tx, current, current);
```

Comment:
Is it correct passing twice `current`, even if signature is `public E update(TxContext txContext, E currentEntity, E updatedEntity`?

Reply:
Yes. The "current" one is used only to copy the creation fields:
![image](https://github.com/eclipse/kapua/assets/115478125/9ed58957-f0c6-414c-9e90-063cc724df89)

So passing the same one, if fetched from the db and not coming from the outside, is a valid case